### DISCUSSION TITLE
Socket.IO connection not working

### DISCUSSION THREAD

**Discussion by joaogbp** [Links2004/arduinoWebSockets#832] [Q&A]
I am trying to connect my ESP32 to a server using socket.io.
The server is working for the socketios's examples (chat and whiteboard), but I can not connect my ESP32 to it.

It is not working.

I tried using ws://my_server and it failed.
I tried using https://my_server and it failed.

Please can anyone help me.

**Comment by joaogbp** [comment]
Can you help me @Links2004?

**Reply by d-r-lenin** [reply]
Is the problem solved now? 

Anyway, I had the same issue. Apparently the socket.io library is not supported on my board. So I had to use the web sockets. In the server use web sockets instead of socket.io if possible or  if you really need socket.io, use both socket.io and web sockets and connect the input/output of them.

**Reply by d-r-lenin** [reply]
If you are using web sockets on the board it will not connect to the socket.io server since socket.io uses a slightly higher level protocol which uses web sockets and not web sockets itself.
